modules/vmfcore/java/FieldDesc.java

```diff
@@ -0,0 +1,34 @@
+//package com.intel.vmf;
+
+public class FieldDesc
+{
+    static
+    {
+        System.loadLibrary("vmf");
+    }
+	
+    public boolean optional;
+	
+    public String name;
+	
+    public int type;
+    
+    public FieldDesc ()
+    {
+    	optional = false;
+    	name = "";
+    	type = Variant.type_string;
+    }
+	
+    public FieldDesc (String newName, int newType, boolean isOptional )
+    {
+    	optional = isOptional;
+    	name = newName;
+    	type = newType;
+    }
+	
+    public boolean isEqual (FieldDesc other )
```

Comment:
Java has a predefined method for objects comparison, you need to overwrite something like:
```java
public boolean equals (Object obj)
  {
     if (this==obj) return true;
     if (obj == null) return false;
     if (this.getClass() != obj.getClass()) return false;
     FieldDesc that = (FieldDesc) obj ;
     return this.optional ==  that.optional && this.type ==  that.type && this.name.equals(that.name);
   }
```